File: mars/src/test/java/TurnRightTest.java
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class TurnRightTest {

    @BeforeEach
    void setUp() {
    }

    @AfterEach
    void tearDown() {
    }

    @Test
    @DisplayName("北 -> 右转")
    void test_TurnRight_excute_S() {
        TurnRight turnRight = new TurnRight();
        Point point = new Point(0, 0, Direction.SOUTH);
        assertEquals(Direction.EAST, turnRight.execute(point).getDirection());
    }

    @Test
    @DisplayName("东 -> 右转")
    void test_TurnRight_excute_E() {
        TurnRight turnRight = new TurnRight();
        Point point = new Point(0, 0, Direction.EAST);
        assertEquals(Direction.NORTH, turnRight.execute(point).getDirection());
    }

    @Test
    @DisplayName("南 -> 右转")
    void test_TurnRight_excute_N() {
        TurnRight turnRight = new TurnRight();
        Point point = new Point(0, 0, Direction.NORTH);
        assertEquals(Direction.WEST, turnRight.execute(point).getDirection());
    }

    @Test
    @DisplayName("西 -> 右转")
    void test_TurnRight_excute_W() {
        TurnRight turnRight = new TurnRight();
        Point point = new Point(0, 0, Direction.WEST);
        assertEquals(Direction.SOUTH, turnRight.execute(point).getDirection());
    }
}

File: mars-java-kotlin/src/test/kotlin/TurnRightTest.kt
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.DisplayName

internal class TurnRightTest {

    @Test
    @DisplayName("测试右转指令")
    fun `test excute`() {
        assertEquals(Direction.EAST, TurnRight().excute(Point(0, 0, Direction.SOUTH)).direction)
        assertEquals(Direction.NORTH, TurnRight().excute(Point(0, 0, Direction.EAST)).direction)
        assertEquals(Direction.WEST, TurnRight().excute(Point(0, 0, Direction.NORTH)).direction)
        assertEquals(Direction.SOUTH, TurnRight().excute(Point(0, 0, Direction.WEST)).direction)
    }


}